lib/rubocop/cop/rspec/factory_bot/factory_class_name.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Use string value when setting the class attribute explicitly.
+        #
+        # @example
+        #   # bad
+        #   factory :foo, class: Foo do
+        #   end
+        #
+        #   # good
+        #   factory :foo, class: 'Foo' do
+        #   end
+        class FactoryClassName < Cop
+          MSG = "Pass '%<class_name>s' instead of %<class_name>s."
+
+          def_node_matcher :class_name, <<~PATTERN
+            (send _ :factory _ (hash <(pair (sym :class) $_) ...>))
```

Comment:
You may rewrite `$_` as `$(const)` and get rid of a guard statement.
It's up to you to decide if this node pattern is becoming overly complex though. I have no objections on keeping it this way.